packages/react-button/src/components/CompoundButton/stories/CompoundButtonToggleable.stories.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import { CalendarMonthRegular } from '@fluentui/react-icons';
+import { useMergedEventCallbacks } from '@fluentui/react-utilities';
+import { mergeClasses } from '@griffel/react';
+import { useToggleState } from '../../../utils/useToggleState';
+import {
+  renderCompoundButton_unstable,
+  useCompoundButton_unstable,
+  useCompoundButtonStyles_unstable,
+} from '../../../CompoundButton';
+import {
+  useCheckedStyles as useToggleButtonCheckedStyles,
+  useDisabledStyles as useToggleButtonDisabledStyles,
+} from '../../ToggleButton/useToggleButtonStyles';
+import type { ForwardRefComponent } from '@fluentui/react-utilities';
+import type { CompoundButtonProps, CompoundButtonState } from '../../../CompoundButton';
+import type { ToggleButtonProps, ToggleButtonState } from '../../../ToggleButton';
+
+type ToggleCompoundButtonProps = CompoundButtonProps & ToggleButtonProps;
+type ToggleCompoundButtonState = CompoundButtonState & ToggleButtonState;
+
+const useToggleCompoundButton = (
+  props: ToggleCompoundButtonProps,
+  ref: React.Ref<HTMLButtonElement | HTMLAnchorElement>,
+): ToggleCompoundButtonState => {
+  const compoundButtonState = useCompoundButton_unstable(props, ref);
+  const { onClick, role } = compoundButtonState.root;
+
+  const { 'aria-checked': ariaChecked, 'aria-pressed': ariaPressed, checked, onClick: onToggleClick } = useToggleState({
```

Comment:
do we have a documentation for which props need to be deconstructed and applied to root? Or can we change this to take all props, apply them to root and only override onClick?

Some people will say that spreading all props is bad practice, but deconstructing them means that consumers need to have understanding of what props are there. If we add a new prop, they will need to go in their code and add it as well - which increases the API area of our hooks.

Reply:
Knowing which props to apply to root is a good point. perhaps some kind of syntax like the following?

```ts
{ props, checked } = useToggleState()
```

However, I don't quite agree with adding props, since that is a similar to what happens when we introduce a new prop to a component. It is a backwards compatible new feature. Users would still need to modify their code to use the new prop. 

But ofc when adding new props there is always the risk of having to maintain it

Reply:
question is - if we add a new prop, how would they know that they need to change something? With complex products, the team that is doing upgrades might not be aware of all the individual features that might use the compound toggle button. It of course depends on how the code is structured on the partner side, but in general, I can imagine situations where it might be hard to have the understanding of what needs to be changed where.

Reply:
I don't think adding a prop would ever mean that we would be breaking or diminishing an existing feature, since that would be a breaking API change right ?

Reply:
Well, removing a prop is a breaking change, adding a prop is not necessarily breaking. The tricky thing here is that everything goes to the root except `checked`, but `onClick` needs to be overridden. We could abstract all of that inside of the hook so that it just returns `checked` and `root` and you have to spread root yourself, but I wonder if at that point we're doing too much (we'd end up with a similar hook to the one @ling1726 initially had feedback on).

Reply:
After some thought I agree with @jurokapsiar  `useToggleState(state)` is a lot more future proof and less ambiguous compared to knowing what to spread and how

Reply:
Went back to the original implementation now.